.github/workflows/lint-test.yml

```diff
@@ -77,20 +77,19 @@ jobs:
 
   lint:
     runs-on: ubuntu-latest
+    env:
+      RUFF_OUTPUT_FORMAT: github
     steps:
+
       - uses: actions/checkout@v4
 
-      - name: Cache pre-commit
-        uses: actions/cache@v4
-        with:
-          key: pre-commit-${{ hashFiles('.pre-commit-config.yaml') }}
-          path: ~/.cache/pip
```

Comment:
Seems to me like this would still be needed here? The action doesn't cache the pip dependency, only the hook repositories